### ISSUE TITLE
Functions Core Tools creates CS triggers in flat structure instead of folder structure that docs recommend

### ISSUE THREAD

**Issue by brandonh-msft** [Azure/azure-functions-core-tools#1140]
Docs and code creation mismatch on folder structure creates some confusion as to whether the functions will work, and if I were a customer I would not want to have to experiment to find out if the way it is done in VS Code will work when the docs say I should expect something different.

In this folder structure doc, it is recommended to create folders for each function inside a function app but in Core Tools the CS functions are created in single files.  Javascript function apps are created as recommended in the docs. Given that the docs reference function.json files and these are not available in the CSharp function app, it may be that the doc should be updated rather than the core tools code. However the folder structure is reinforced in both the C# (class library) doc and the C# script (.csx) doc. Oddly, all of these docs reference a functions.json file but Core Tools creates the functions.json file in the \bin\Debug\netcoreapp2.1\ folder (see attached screenshot).

**Repro step**: Create multiple CSharp triggered functions in the same folder using Core Tools in VS Code.

**Problem Statement**: Currently, using Core Tools in VS Code creates CS Functions in single files. The documentation recommends that each function be created inside of a function app, and the JavaScript Functions are created following this recommendation, so the partner was confused why CS Functions do not follow this.

**Desired Experience**: When using Core Tools in VS Code to create CS Functions, the resulting function should be created inside of a function app.

<details><summary>Internal tracking</summary>#191161</details>

**Comment by ahmelsayed** [comment]
The folder structure has no significance for C#, so the I didn't think I'd create a folder by default. It looks like VS and VS code both do however. I'll update the cli to create a folder per function to match VS behavior.

> Oddly, all of these docs reference a functions.json file but Core Tools creates the functions.json file in the \bin\Debug\netcoreapp2.1\ folder (see attached screenshot).

function.json is supposed to be a build artifact for C# functions, not something that you need to see or deal with. They are autogenerated for the rest of the tooling to use based on the attributes you have in your C# code.

**Comment by brandonh-msft** [comment]
awesome, thanks @ahmelsayed!

cc @theangellmethod

**Comment by theangellmethod** [comment]
@ahmelsayed based on your comment about functions.json for C#, would you agree that it is appropriate to update the documentation to highlight that for C# (as opposed to Node) the user should not look at modifying/interacting with this file?

**Comment by ahmelsayed** [comment]
@theangellmethod yes, I think that would make sense. This is also true for Java btw. Though I'm not sure where in the docs to add that.
